compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala

```diff
@@ -214,9 +214,25 @@ class ErrorMessagesTests extends ErrorMessagesTest {
       val defn = ictx.definitions
 
       assertMessageCount(1, messages)
-      val OverloadedOrRecursiveMethodNeedsResultType(tree) :: Nil = messages
-      assertEquals("foo", tree.show)
-    }
+      val OverloadedOrRecursiveMethodNeedsResultType(treeName) :: Nil = messages
+      assertEquals("foo", treeName)
+    }
+
+//The following test is disabled (may be running in an unsupported phase)
+//
+//    checkMessagesAfter("frontend") {
+//      """
+//        |case class Foo[T](x: T)
+//        |object Foo { def apply[T]() = Foo(null.asInstanceOf[T]) }
+//      """.stripMargin
+//    }.expect { (ictx, messages) =>
+//      implicit val ctx: Context = ictx
+//
+//      assertMessageCount(1, messages)
+//      val OverloadedOrRecursiveMethodNeedsResultType(treeName2) :: Nil = messages
+//      assertEquals("Foo", treeName2)
+//    }
+  }
```

Comment:
I cloned your branch and tried out the commented out test - and it runs fine for me. Did you do `git submodule update --init`?